### PR TITLE
Fix etcd storage location for CRs

### LINKED ISSUE #46031: api-extensions etcd storage:  resources serialized in etcd with double prefix
@deads2k as discussed in #46030  more particulary [here](https://github.com/kubernetes/kubernetes/pull/46030#issuecomment-302411867)
CRD are serialized in`etcd` with a double prefix.
Thanks to have a look

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/kubernetes/kubernetes/issues/46031

The first commit is easy, the second commit is hard.  

@sttts Read the second commit very carefully and search the repo to make sure that you agree this is only affecting aggregator and apiextensions.

@pmorie @derekwaynecarr make sure you check to see if service catalog is double prefixing.  Looks like we missed it in 1.6, but no one was using it.

### CODE REVIEW

**Comment by deads2k** [comment]
approved, linked issue.

**Comment by liggitt** [comment]
in 1.6:
* simpleRestOptionsFactory is only referenced in EtcdOptions.ApplyTo()
* Etcd.ApplyTo() is only referenced in RecommendedOptions.ApplyTo()
* RecommendedOptions is only used in kube-aggregator and sample-apiserver
* as a sanity check, I put a panic in simpleRestOptionsFactory.GetRESTOptions and kube-apiserver starts up just fine

**Comment by deads2k** [comment]
> in 1.6:
> 
> simpleRestOptionsFactory is only referenced in EtcdOptions.ApplyTo()
> Etcd.ApplyTo() is only referenced in RecommendedOptions.ApplyTo()
> RecommendedOptions is only used in kube-aggregator and sample-apiserver
> as a sanity check, I put a panic in simpleRestOptionsFactory.GetRESTOptions and kube-apiserver starts up just fine

Ok, then I think we're good to merge the change as proposed here without any compensation.  I've already tagged in the service catalog guys who I know were using this early.

File: staging/src/k8s.io/apiserver/pkg/server/options/etcd.go [hunk 1]
```diff
@@ -135,7 +135,7 @@ func (f *SimpleRestOptionsFactory) GetRESTOptions(resource schema.GroupResource)
 		Decorator:               generic.UndecoratedStorage,
 		EnableGarbageCollection: f.Options.EnableGarbageCollection,
 		DeleteCollectionWorkers: f.Options.DeleteCollectionWorkers,
-		ResourcePrefix:          f.Options.StorageConfig.Prefix + "/" + resource.Group + "/" + resource.Resource,
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd.go]: ignoring the prefix in general feels strange. If it is empty, it is what I would expect. But if I set `StorageConfig.Prefix` to non-empty?

Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd.go]: I understand from the linked issue discussion that we actually had a double-prefix. Do we have a test that we now have a single-prefix? Then I could agree.

Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd.go]: > I understand from the linked issue discussion that we actually had a double-prefix. Do we have a test that we now have a single-prefix? Then I could agree.

Yes, the one modified in this pull.

Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd.go]: > ignoring the prefix in general feels strange. If it is empty, it is what I would expect. But if I set StorageConfig.Prefix to non-empty?

StorageConfig (which is preserved) sets the prefix.

**Comment by liggitt** [comment]
@k8s-bot pull-kubernetes-bazel  test this

**Pull request merged**